src/main/java/com/conveyal/gtfs/stats/StopStats.java

```diff
@@ -1,7 +1,239 @@
 package com.conveyal.gtfs.stats;
 
+import com.conveyal.gtfs.GTFSFeed;
+import com.conveyal.gtfs.model.Route;
+import com.conveyal.gtfs.model.Service;
+import com.conveyal.gtfs.model.StopTime;
+import com.conveyal.gtfs.model.Trip;
+import com.conveyal.gtfs.stats.model.TransferPerformanceSummary;
+import gnu.trove.list.TIntList;
+import gnu.trove.list.array.TIntArrayList;
+import org.mapdb.Fun;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
 /**
  * Created by landon on 9/2/16.
  */
 public class StopStats {
+
+    private GTFSFeed feed = null;
+    private FeedStats stats = null;
+    private RouteStats routeStats = null;
+
+    public StopStats (GTFSFeed f, FeedStats fs) {
+        feed = f;
+        stats = fs;
+        routeStats = stats.route;
+    }
+
+    public Set<Route> getRoutes (String stop_id) {
+        return feed.patterns.values().stream()
+                .filter(p -> p.orderedStops.contains(stop_id))
+                .map(p -> feed.routes.get(p.route_id))
+                .collect(Collectors.toSet());
+    }
+
+    public int getRouteCount (String stop_id) {
+        return getRoutes(stop_id).size();
+    }
+
+    public int getTripCountForDate (String stop_id, LocalDate date) {
+        return (int) getTripsForDate(stop_id, date).size();
+    }
+
+    /** Get list of trips for specified date of service */
+    public List<Trip> getTripsForDate (String stop_id, LocalDate date) {
+        List<String> tripIds = stats.getTripsForDate(date).stream()
+                .map(trip -> trip.trip_id)
+                .collect(Collectors.toList());
+
+        return feed.getStopTimesForStop(stop_id).stream()
+                .map(t -> feed.trips.get(t.b.a)) // map to trip ids
+                .filter(t -> tripIds.contains(t.trip_id)) // filter by trip_id list for date
+                .collect(Collectors.toList());
+    }
+
+    public int getAverageHeadwayForStop (String stop_id, LocalDate date, LocalTime from, LocalTime to) {
+        List<Trip> tripsForStop = feed.getStopTimesForStop(stop_id).stream()
+                .map(t -> feed.trips.get(t.b.a))
+                .filter(trip -> feed.services.get(trip.service_id).activeOn(date))
+                .collect(Collectors.toList());
+
+        return getStopHeadwayForTrips(stop_id, tripsForStop, from, to);
+    }
+
+    /** Get the route headway for a given service date at a stop over a time window, in seconds */
+    public Map<String, Integer> getRouteHeadwaysForStop (String stop_id, LocalDate date, LocalTime from, LocalTime to) {
```

Comment:
Does this need to accomodate direction_ids? we're going to have biased results otherwise when we compare headways of (generally) bus service with separate stops per direction, and rail/busway service (which generally has one stop for both directions).